neo/SmartContract/ApplicationEngine.cs

```diff
@@ -81,6 +84,27 @@ protected override bool PreExecuteInstruction()
         {
             if (CurrentContext.InstructionPointer >= CurrentContext.Script.Length)
                 return true;
+            
+            Instruction instruction = CurrentContext.CurrentInstruction;
+            switch(instruction.OpCode)
+            {
+                case OpCode.JMP:
+                case OpCode.JMPIF:
+                case OpCode.JMPIFNOT:
+                case OpCode.CALL:
```

Comment:
With syscall, you can jump to your own contract again

Reply:
From what we've been talking, most syscalls will be disabled on verification. In my opinion now, only crypto stuff and block header information should be available... no appcalls, no storage, ... it will be fast as a bullet and state independent.